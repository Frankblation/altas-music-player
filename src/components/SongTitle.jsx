import React from "react"

export default function SongTitle({ title, artist }) {
  return (
    <div className="text-center">
      <h2 className="text-2xl font-bold truncate">{title}</h2>
      <p className="text-gray-600 truncate">{artist}</p>
    </div>
  )
}

