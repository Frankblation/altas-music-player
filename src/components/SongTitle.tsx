

const SongTitle = ({ title, artist }) => {
  return (
    <div className="text-center">
      <h2 className="text-2xl font-bold text-gray-800">{title}</h2>
      <p className="text-lg text-gray-600">{artist}</p>
    </div>
  )
}

export default SongTitle

