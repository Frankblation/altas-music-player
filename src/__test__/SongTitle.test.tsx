import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import SongTitle from "../components/SongTitle";

test("SongTitle renders correctly with title and artist", () => {
  const { container } = render(<SongTitle title="Bohemian Rhapsody" artist="Queen" />);
  expect(container).toMatchSnapshot();
});

test("SongTitle renders correctly with only title", () => {
  const { container } = render(<SongTitle title="Imagine" artist="" />);
  expect(container).toMatchSnapshot();
});
