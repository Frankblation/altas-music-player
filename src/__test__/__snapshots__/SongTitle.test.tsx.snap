// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle renders correctly with only title 1`] = `
<div>
  <div
    class="text-center"
  >
    <h2
      class="text-2xl font-bold text-gray-800"
    >
      Imagine
    </h2>
    <p
      class="text-lg text-gray-600"
    />
  </div>
</div>
`;

exports[`SongTitle renders correctly with title and artist 1`] = `
<div>
  <div
    class="text-center"
  >
    <h2
      class="text-2xl font-bold text-gray-800"
    >
      Bohemian Rhapsody
    </h2>
    <p
      class="text-lg text-gray-600"
    >
      Queen
    </p>
  </div>
</div>
`;
